proto/iavl_api.proto

```diff
@@ -0,0 +1,252 @@
+syntax = "proto3";
+package iavl;
+
+option go_package = "proto/iavl";
```

Comment:
I'm not super familiar with the combination of `go` plugins for `protoc` that glue everything, together with the bash script. I was kind of just stumbling around until I got something to work -- it was only non-trivial because this package imports something from `proof.proto`. If I can still give `proof.proto` and `iavl_api.proto` the same `go_package`, I don't care what it's called. Otherwise I would need someone to explicitly tell me the file structure, package names that you want.

On a side note, there are too many `*proto*` packages/libraries/plugins in go :exploding_head:

Reply:
I think we should move all of the Protobuf schemas into the same file, and use `proto` as the Go package name.

>On a side note, there are too many *proto* packages/libraries/plugins in go 🤯

Tell me about it, we've wasted a significant amount of time dealing with the fallout of this lately.

Reply:
Not sure how helpful this is, but for reference of how we organize proto packages in Cosmos SDK, you can see [this ADR](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-023-protobuf-naming.md) which documents our guidelines for proto package & naming conventions.

and here's the PR that tackled this for the SDK - https://github.com/cosmos/cosmos-sdk/pull/6413 

Multiple files can indeed have the same `go_package`'s. I'm not too familiar w/ how IAVL is already structuring its proto stuff though, and I'd default to @erikgrinaker's suggestions so that we have internal consistency with how IAVL is already doing proto stuff for now.

Reply:
Following the SDK's guidelines makes sense; we only have a handful of internal Protobuf messages currently.